### PR TITLE
Added regression model and base training function 

### PR DESCRIPTION
**Pull request by kaiserliche**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Relevant Issue
Issue #90 

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/NimbleEdge/RecoEdge/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/NimbleEdge/RecoEdge/blob/main/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [NimbleEdge Styleguide](https://numpydoc.readthedocs.io/en/latest/format.html)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/NimbleEdge/RecoEdge/labels)
- [ ] My changes are covered by tests

### CODE REVIEW

File: experiments/basic_func.py [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+from fedrec.utilities import registry
+from fedrec.utilities import saver_utils as saver_mod
+from fedrec.utilities.cuda_utils import map_to_cuda
+from fedrec.utilities.logger import BaseLogger
+from sklearn import metrics
+from tqdm import tqdm
+
+from fedrec.utilities.random_state import Reproducible
+
+class Basic:
```
Comment (ramesht007) [inline experiments/basic_func.py:15]: change class name to `base trainer`
Change module name to base_trainer.py

File: experiments/basic_func.py [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+        self._data_loaders = {}
+    # @staticmethod
+    def _yield_batches_from_epochs(loader, start_epoch):
+        current_epoch = start_epoch
+        while True:
+            for batch in loader:
+                yield batch, current_epoch
+            current_epoch += 1
+
+    # @property
```
Comment (ramesht007) [inline experiments/basic_func.py]: remove @property

File: experiments/dlrm/dlrm_trainer.py [hunk 3]
```diff
@@ -14,6 +15,8 @@
 from fedrec.utilities.random_state import Reproducible
 
 
+from experiments.basic_func import Basic
+
 @attr.s
 class DLRMTrainConfig:
```
Comment (ramesht007) [inline experiments/dlrm/dlrm_trainer.py:22]: add `base trainer` as parent class

**Review by ramesht007 — COMMENTED** [review]
- Add `base_trainer` as parent class for  ```class RegressionTrainer```

File: experiments/__init__.py [hunk 4]
```diff
@@ -1 +1 @@
-from . import dlrm
\ No newline at end of file
+from . import dlrm
```
Comment (vkkhare) [inline experiments/__init__.py:1]: add regression here as well

File: experiments/basic_func.py (hunk 1 above)
Comment (vkkhare) [inline experiments/basic_func.py:15]: Move this fedrec

File: experiments/regression/reg_trainer.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+    # @staticmethod
+    Basic._yield_batches_from_epochs(loader, start_epoch)
+
+    # @property
+    Basic.model(self)
+
+    # @property
+    Basic.optimizer(self)
+    Basic.get_scheduler(self, optimi, **kwargs)
```
Comment (vkkhare) [inline experiments/regression/reg_trainer.py:79]: You do not call static methods of the super class. Instead subclass with the super class and use the functions as normal

**Review by vkkhare — CHANGES_REQUESTED** [review]

**Pull request closed without merging**